### DISCUSSION TITLE
XBackBone 4.0 wishlist

### DISCUSSION THREAD

**Discussion by sergix44** [SergiX44/XBackBone#404] [Ideas]
Hello everyone! I was thinking as per title about a partial rewrite of xbackbone, and i would like to know what features you would like to have included, just to have an idea.
Thanks!

**Comment by Facinorous-420** [comment]
It would be cool to have zero width spacing for url naming for when sharing the link.

**Comment by 6K6666** [comment]
embed videos?

**Comment by runarsf** [comment]
A way of customizing the user ID (not just to a 5-character string) without editing the database would be nice to have.
And a way of uploading files without a user-ID at all, preferably with the ability to enable public uploading.

**Comment by EricGuic** [comment]
The way to manage multiple resolution for video file ? (as mentionned here : https://github.com/SergiX44/XBackBone/discussions/383)
Thanks ;-)

**Comment by Tinoisthe** [comment]
1st Public and Private files should have own tab area .

make it so people don't need a account to upload

**Comment by gramakri** [comment]
Upload from clipboard would be useful for those not on windows (sharex).

**Comment by Nimfek** [comment]
- Multi domain support (I have few of them for few projects - users, that are using XBackBone on my server). Maybe as an option in user profile edditing?
- Option to change user tag directly from admin panel.
- Theme change on user side and one default one.
- Little notepad (pastebin like)
- Filter option - seeing other users files as an admin is really anoying. I would like an option to turn this off.
- Preview of image / gif / video on mouse hover

**Comment by EntraptaJ** [comment]
OpenID Connect/OAuth2 SSO alongside LDAP sync.

My Identity provider does support LDAP and I have it tied into SSO that way, however using standard OIDC/OAUTH would make the user experience better.

**Reply by gramakri** [reply]
Would be great to have OIDC!

**Reply by DavidFuchs** [reply]
OIDC is number 1 for me. I want to have all my homelab services integrated with Authentik.

And if not OIDC integration, then at least the ability to set an auto-login user for the web UI so I can bypass auth.

**Comment by ZingyAwesome** [comment]
oEmbed support and ability to customise embeds without having to edit src.

**Comment by skullz613** [comment]
Just been playing around with this and would like to see a option to add a background image that matches my sites one.
I also would like to have catagories with folders etc... 
Also a way to block extensions not wanted

**Comment by Tampa** [comment]
I always kinda liked how imgur did albums, any chance for such a feature?

**Comment by stb109** [comment]
At rest encryption of the images?

**Comment by nunawinter** [comment]
upload from remote server and url would be excellent .
